### PR TITLE
applied rpath fixes for  linux

### PR DESCRIPTION
**Pull request by cekees**
@certik all I did here is copy your approach in the rpath_openssl branch to the newer version of openssl.yaml, which is in openssl/openssl.yaml and applies a patch, etc.

### CODE REVIEW

File: pkgs/openssl/openssl.yaml [hunk 1]
```diff
@@ -23,13 +23,20 @@ build_stages:
...
+  handler: bash
+  bash: |
+    export OPENSSL_EXTRA_FLAGS="-Wl,-rpath=$ARTIFACT/lib -Wl,-rpath=$ZLIB_DIR/lib"
+
 - name: configure
   after: fix_docs
   handler: bash
   mode: replace
   bash: |
     #./config --prefix=$ARTIFACT shared zlib-dynamic -I$ZLIB_DIR/include -Wl,-rpath=$ARTIFACT/lib -Wl,-rpath=$ZLIB_DIR/lib
```
Comment (certik) [inline pkgs/openssl/openssl.yaml]: Let's remove this comment then.

**Comment by certik** [comment]
This looks good to me. However, before we merge, can you post `ldd` commands from before and after to verify that this fixes anything, at least on your machine? I used my current `openssl` installation and I am not able to find a problem that needs fixing:
```
certik@redhawk:/local/certik/bld/openssl/pkrr3ils6o2k/lib/engines$ ldd libgost.so 
	linux-vdso.so.1 =>  (0x00007fff2a1ff000)
	libcrypto.so.1.0.0 => not found
	libc.so.6 => /lib64/libc.so.6 (0x00007f076df02000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f076e4cc000)
```
Except the not found library ---- how does that manifest? How can I trigger the linker failure?

I wasn't able to find any linking to my systemwide zlib library either.

**Comment by cekees** [comment]
before:

```
~/.h/b/o/p/l/engines ❯❯❯ ldd libgost.so
	linux-vdso.so.1 =>  (0x00007fff87996000)
	libcrypto.so.1.0.0 => /lib/x86_64-linux-gnu/libcrypto.so.1.0.0 (0x00007f3e758ed000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f3e75528000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f3e75323000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f3e75f12000)
```

after

```
~/.h/b/o/w/l/engines ❯❯❯ ldd libgost.so
	linux-vdso.so.1 =>  (0x00007ffc101e2000)
	libcrypto.so.1.0.0 => /home/cekees/.hashdist/bld/openssl/wk6dzdap4su2/lib/libcrypto.so.1.0.0 (0x00007f7149f19000)
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f7149b25000)
	libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f7149921000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f714a51c000)
```

**Comment by certik** [comment]
Cool, that works. +1 to merge.

**Comment by cekees** [comment]
closes #57

**Pull request merged**